### PR TITLE
Add single source single compiler pass (SSCP) to the glossary

### PR DESCRIPTION
**Pull request by keryell**
Also add some links back to the subsections of SMCP and SSCP.

### CODE REVIEW

**Review by tomdeakin — APPROVED** [review]

File: adoc/chapters/glossary.adoc [hunk 1]
```diff
@@ -432,14 +432,16 @@ object. For the full description please refer to <<subsec:buffers>>.
...
+[[smcp]]SMCP:: The single-source multiple compiler-passes (SMCP)
+    technique allows a single-source file to be parsed by multiple
+    compilers for building native programs per compilation target. For
+    example, a standard {cpp} CPU compiler for targeting <<host>> will
+    parse the <<sycl-file>> to create the {cpp} <<sycl-application>>
+    which offloads parts of the computation to other
+    <<device,devices>>. A SYCL device compiler will parse the same
+    source file and target only SYCL kernels. For the full description
+    please refer to <<subsec:smcp>>.  See <<sscp>> for the dual
+    approach.
```
Comment (gmlueck) [inline adoc/chapters/glossary.adoc:444]: I'm not sure about the last sentence.  Should we say:

> See \<\<sscp\>\> for another approach.

?

File: adoc/chapters/glossary.adoc [hunk 2]
```diff
@@ -451,6 +453,16 @@ object. For the full description please refer to <<subsec:buffers>>.
     the value prior to the compilation of a <<kernel-bundle>> and in a
     <<sycl-kernel-function>> for retrieving the value during invocation.
 
+[[sscp]]SSCP:: The single-source single compiler-pass (SSCP) technique
+    allows a single-source file to be parsed only once by a single
+    compiler. For example, the SYCL compiler will parse the
+    <<sycl-file>> once. Then, from this single intermediate
+    representation, for each kind of device architecture a compilation
+    flow will be generate the binary for each kernel and another
```
Comment (gmlueck) [inline adoc/chapters/glossary.adoc:461]:
```suggestion
    flow will generate the binary for each kernel, and another
```

File: adoc/chapters/glossary.adoc (hunk 1 above)
Comment (keryell) [reply adoc/chapters/glossary.adoc:444]: You are right. Let's not use pseudo-mathematical jargon here.

**Review by gmlueck — APPROVED** [review]

**Review by illuhad — APPROVED** [review]

**Comment by fraggamuffin** [comment]
lgtm

**Pull request merged**